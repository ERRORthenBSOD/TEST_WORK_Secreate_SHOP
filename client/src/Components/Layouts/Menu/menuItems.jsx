import React from 'react';
import {
    Divider,
    List,
    ListItem,
    ListItemIcon,
    ListItemText,
    Typography,
} from '@material-ui/core';


import {
    ExitToApp,
    AddShoppingCart,
    ShoppingCart
} from '@material-ui/icons';





const menuItems = (toggleDrawer, classes, logout) => (
    <div
        className={classes.list}
        role="presentation"
    >
        <Typography variant="h4" color='primary' className={classes.title}>
            Меню
        </Typography>
        <Divider />
        <List className={classes.menuList}>
             <ListItem button onClick={toggleDrawer(false, 'shop')}>
                <ListItemIcon>
                    <AddShoppingCart/>
                </ListItemIcon>
                <ListItemText primary={
                    <Typography className={classes.listTitle}>Магазин</Typography>
                }/>
            </ListItem>
            <Divider />
        </List>
        <List className={classes.menuList}>
            <ListItem button onClick={toggleDrawer(false, 'cart')}>
                <ListItemIcon>
                    <ShoppingCart/>
                </ListItemIcon>
                <ListItemText primary={
                    <Typography className={classes.listTitle}>Корзина покупок</Typography>
                }/>
            </ListItem>
            <Divider />
        </List>
        <Divider />
        <List>
            <ListItem button onClick={logout}>
                <ListItemIcon>
                        <ExitToApp />
                </ListItemIcon>
                <Typography className={classes.listTitle}>Выход</Typography>
            </ListItem>
        </List>
    </div>
);

export default menuItems;