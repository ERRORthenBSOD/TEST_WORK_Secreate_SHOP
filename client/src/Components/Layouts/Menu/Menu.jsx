import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import {withRouter} from 'react-router-dom'
import menuItems from './menuItems'


import {SwipeableDrawer,} from '@material-ui/core';
import PropTypes from "prop-types";


const useStyles = makeStyles(theme => (
    {
        list: {
            width: 400,
            color: theme.palette.secondary.main,
        },
        fullList: {
            width: 'auto',

        },
        title: {
            textAlign: 'center',
            color: theme.palette.secondary.main,
            padding: 10,
        },
        heading: {
            fontSize: theme.typography.pxToRem(15),
            // flexBasis: '33.33%',
            flexShrink: 0,
            fontWeight: 600
        },
        secondaryHeading: {
            fontSize: theme.typography.pxToRem(15),
            color: theme.palette.text.secondary,
        },
        menuList: {
            padding: 0,
            ...theme.typography.button,
            backgroundColor: theme.palette.background.paper,
        },
        listTitle: {
            fontWeight: 600
        },
        noPadding: {
            padding: '0 0 0 24px'
        }
    }
));

function Menu(props) {
    const classes = useStyles();
    const {toggleDrawer, menuOpened, logout} = props;

    return (
        <>
            <SwipeableDrawer
                open={menuOpened}
                onClose={toggleDrawer(false)}
                onOpen={toggleDrawer(true)}
            >
                {menuItems(toggleDrawer, classes ,logout)}
            </SwipeableDrawer>
        </>
    );
}

export default withRouter(Menu);

Menu.propTypes = {
    toggleDrawer: PropTypes.func,
    menuOpened: PropTypes.bool,
    logout: PropTypes.func,
    history: PropTypes.object,
};