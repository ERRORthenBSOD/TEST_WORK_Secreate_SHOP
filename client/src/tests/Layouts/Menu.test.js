import {shallow} from "enzyme";
import Menu from "../../Components/Layouts/Menu/Menu";
import React from "react";

it('Menu renders without crashing', () => {
    const wrapper = shallow(<Menu />);
    expect(wrapper).toMatchSnapshot()
});
