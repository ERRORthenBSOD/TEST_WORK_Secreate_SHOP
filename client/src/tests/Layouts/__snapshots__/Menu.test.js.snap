// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
